scripts/duplicate_sequencing_group.py

```diff
@@ -0,0 +1,855 @@
+        if dry_run:
+            logger.info(
+                f'DRY RUN :: Skipping analysis creation for {analysis["type"]} analysis.'
+            )
+            continue
+        analysis_to_create = Analysis(
+            type=analysis['type'],
+            status=AnalysisStatus('completed'),
+            outputs={
+                'basename': analysis['output']
+            },  # this is the correct syntax for autofilling outputs dict
+            meta=analysis['meta'],  # add some provenance info here?
+            sequencing_group_ids=[analysis['sg_id']],
+        )
+        promises.append(analysis_api.create_analysis_async(dataset, analysis_to_create))
+
+    return asyncio.gather(*promises)
+
+
+# ============================================================================
+# FILE UTILITY FUNCTIONS
+# ============================================================================
+
+
+def file_size(path: Path) -> int:
+    """Return the size (in bytes) of the object, which is assumed to exist"""
+    if path.exists():
+        return path.stat().st_size
+    return 0
+
+
+def file_reheaderable(path: str) -> bool:
+    """Returns whether this is a file that can be reheadered with samtools/bcftools"""
+    return any(
+        path.endswith(e)
+        for e in [
+            '.cram',
+            '.vcf.gz',
+            '.vcf.bgz',
+            '.g.vcf.gz',
+        ]
+    )
+
+
+def file_gatk_check(path: str):
+    """
+    Check if this is a file from GATK-SV CollectSVEvidence that needs bgzip reheadering,
+    and return the (1-based) column index of the sample ID to replace.
+    """
+    file_dict = {
+        # paired-end evidence (sample ID in column 3)
+        # chr1    10000   SAMPLE_ID       174     0       0       3
+        '.sd.txt.gz': 3,
+        # split-read evidence (sample ID in column 5)
+        # chr1    10000   left    4       SAMPLE_ID
+        '.sr.txt.gz': 5,
+        # discordant paired-end evidence (sample ID in column 7)
+        # chr1    10000   -       chr5    10495   +       SAMPLE_ID
+        '.pe.txt.gz': 7,
+    }
+    for suffix, sample_id_col in file_dict.items():
+        if path.endswith(suffix):
+            return sample_id_col
+    return None
+
+
+# ============================================================================
+# BATCH PROCESSING AND FILE REHEADERING FUNCTIONS
+# ============================================================================
+
+
+async def reheader_analysis_files_in_batch(
+    source_sequencing_group_id: str,
+    new_sequencing_group_id: str,
+    source_paths_new_paths: list[tuple[Path, Path]],
+):
+    """
+    Reheader analysis files in Hail Batch to update the sequencing group IDs within the file headers.
+
+    Uses samtools for CRAM files and bcftools for VCF/GVCF files.
+    For other text based files, performs a simple find-and-replace operation with sed.
+    """
+    batch = get_batch()
+    for old_path, new_path in source_paths_new_paths:
```

Comment:
can we insert an existence check here on the new file path? It's possible that a previous transfer partially completed, in which case the destination file may already exist